src/writer.rs

```diff
@@ -386,6 +386,13 @@ impl FdtWriter {
             .try_into()
             .map_err(|_| Error::TotalSizeTooLarge)?;
 
+        let totalsize = self
+            .data
+            .len()
+            .checked_add(self.strings.len())
+            .ok_or(Error::TotalSizeTooLarge)?;
+        let totalsize = totalsize.try_into().map_err(|_| Error::TotalSizeTooLarge)?;
```

Comment:
Just as a remark/point of view, I think that code is clearer when using `T::try_from` instead of `x.try_into`, just so it's easier for the reader to see what type are we converting into, and also ensures we don't actually convert into something else by accident.

Reply:
We're using try_into throughout the code. I generally agree with what you are saying, but I don't think it applies here because we call into a function which receives `u32` as a parameter. I'll leave it as is for now.